genomic-medicine-sweden/JASEN
# Convert deployment scripts to Makefile(s)

# Description

Due to problems with the installation, where the script for downloading and prepping databases etc would stall multiple times, requiring a Ctrl-C to continue, as reported in #163 , which would leave us as the user unsure if the installation completed, the idea came up to write the deployment script as Makefiles instead, as this would:

1. Make the installation script more modular.
2. Make it possible to resume a failed process.
3. Make it possible to delete and update only some databases later.

This PR does the following:

- Replaces the `deploy/deploy_references_singularity.sh` script with a `Makefile` in the repo root.
- Replaces the `container/build_container.sh` script with a `Makefile` in `/container`, which is called from the main Makefile as well.

As Makefile syntax might be new to some of you, I included a brief description of the syntax in the main Makefile. Otherwise, also see the [official doc(s)](https://www.gnu.org/software/make/manual/make.html).

## Primary function of PR

# Testing

To thoroughly test this:

1. Check out this branch in a completely new folder:
   ```
   git clone -b 164-deployment-script-as-makefile git@github.com:genomic-medicine-sweden/JASEN.git`
   ```
2. Enter the directory:
   ```
   cd JASEN
   ```
3. Run the installation:
   ```
   make install
   ```
4. Run the checks:
   ```
   make check
   ```
   
If you want to only check the container build, then do this:

```
cd JASEN/container
make all
```

# Sign-offs
- [x] Code reviewed by @Emkago 
- [x] Code reviewed by @ryanjameskennedy 
- [x] Code tested by @Emkago
